src/instructlab/cli/model/evaluate.py

```diff
@@ -155,16 +176,38 @@ def evaluate(
     merge_system_user_message,
     backend,
     judge_backend,
-    tls_insecure,  # pylint: disable=unused-argument
-    tls_client_cert,  # pylint: disable=unused-argument
-    tls_client_key,  # pylint: disable=unused-argument
-    tls_client_passwd,  # pylint: disable=unused-argument
+    tls_insecure,
+    tls_client_cert,
+    tls_client_key,
+    tls_client_passwd,
     enable_serving_output,
+    input_questions,
+    output_file_formats,
+    system_prompt,
+    temperature,
 ) -> None:
     """Evaluates a trained model"""
     try:
+        # ensure judge_model and output_dir are not None and defaults are
+        # set correctly depending on benchmark being run.
+        if benchmark == Benchmark.MT_BENCH:
+            if judge_model is None:
+                judge_model = ctx.obj.config.evaluate.mt_bench.judge_model
+            if output_dir is None:
+                output_dir = ctx.obj.config.evaluate.mt_bench.output_dir
+        elif benchmark == Benchmark.MT_BENCH_BRANCH:
+            if judge_model is None:
+                judge_model = ctx.obj.config.evaluate.mt_bench_branch.judge_model
+            if output_dir is None:
+                output_dir = ctx.obj.config.evaluate.mt_bench_branch.output_dir
+        elif benchmark == Benchmark.DK_BENCH:
+            if judge_model is None:
+                judge_model = ctx.obj.config.evaluate.dk_bench.judge_model
+            if output_dir is None:
+                output_dir = ctx.obj.config.evaluate.dk_bench.output_dir
+
         evaluate_model(
-            ctx,
+            ctx.obj.config.serve,
```

Comment:
this is ... better, but still, we want to avoid passing entire custom instances of a class to non-cli packages. But, this isn't part of your PR. I will make an issue for this

cc @alinaryan

Reply:
Absolutely. When you look deeper under the hood, what's needed from the serve config isn't much but refactoring that is out of scope for this PR.

Reply:
that should be prioritized asap for api work